pkg/api/configsync/v1alpha1/resource_override.go

```diff
@@ -74,6 +74,10 @@ type OverrideSpec struct {
 	// +listMapKey=containerName
 	// +optional
 	LogLevels []ContainerLogLevelOverride `json:"logLevels,omitempty"`
+
+	// gitSyncOverride specify git-sync container override
+	// +optional
+	GitSyncOverride GitSyncOverride `json:"gitSyncOverride,omitempty"`
```

Comment:
Has this API been discussed in a design review already? I have a few concerns:

1. There's already a `gitSyncDepth` field at this level. Do we need a nested `gitSyncOverride` struct? This introduces some API inconsistency.
2. Does `timeout` need to be specific to `git-sync`? Can this concept be generalized to apply to the other *-sync containers as well with a generic API field?